e2e/tests/plugins/notebook/restrictedNotebook.e2e.spec.js

```diff
@@ -0,0 +1,227 @@
+ * as represented by the Administrator of the National Aeronautics and Space
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+const { test, expect } = require('@playwright/test');
+const path = require('path');
+
+const TEST_TEXT = 'Testing text for entries.';
+const TEST_TEXT_NAME = 'Test Page';
+const CUSTOM_NAME = 'CUSTOM_NAME';
+const COMMIT_BUTTON_TEXT = 'button:has-text("Commit Entries")';
+const SINE_WAVE_GENERATOR = 'text=Unnamed Sine Wave Generator';
+const DROP_AREA = '.c-notebook__drag-area';
+
+async function enterTextEntry(page) {
+    // Click .c-notebook__drag-area
+    await page.locator(DROP_AREA).click();
+
+    // enter text
+    await page.locator('div.c-ne__text').click();
+    await page.locator('div.c-ne__text').fill(TEST_TEXT);
+    await page.locator('div.c-ne__text').press('Enter');
+
+    return;
+}
+
+async function dragAndDropEmbed(page) {
+    // Click button:has-text("Create")
+    await page.locator('button:has-text("Create")').click();
+    // Click li:has-text("Sine Wave Generator")
+    await page.locator('li:has-text("Sine Wave Generator")').click();
+    // Click form[name="mctForm"] >> text=My Items
+    await page.locator('form[name="mctForm"] >> text=My Items').click();
+    // Click text=OK
+    await page.locator('text=OK').click();
+    // Click text=Open MCT My Items >> span >> nth=3
+    await page.locator('text=Open MCT My Items >> span').nth(3).click();
+    // Click text=Unnamed CUSTOM_NAME
+    await Promise.all([
+        page.waitForNavigation(),
+        page.locator('text=Unnamed CUSTOM_NAME').click()
+    ]);
+
+    await page.dragAndDrop(SINE_WAVE_GENERATOR, DROP_AREA);
+
+    return;
+}
+
+async function lockPage(page) {
+    const commitButton = page.locator(COMMIT_BUTTON_TEXT);
+    await commitButton.click();
+
+    // confirmation dialog click
+    await page.locator('text=Lock Page').click();
+
+    return;
+}
+
+async function openContextMenuRestrictedNotebook(page) {
+    // Click text=Open MCT My Items >> span >> nth=3
```

Comment:
Can you update the comment with the name of the action it's intending to click? We have some work to do with our locators and until then, we'll just have to leave comments with the name of the action

Reply:
Done!